### DISCUSSION TITLE
nushell linting and validation tool

### DISCUSSION THREAD

**Discussion by emailstorbala** [nushell/nushell#11958] [Q&A] [answered]
Is there any linting or validation tool to run against a nushell script. I am aware of the tool `nu-check` but it is not of any help in validating my scripts.

I am looking for a tool like `shellcheck` for bash scripts. Is there any tools available or any plans for that?

**Comment by fdncred** [comment, marked as answer]
Nothing like this exists yet. Just nu-check.

**Reply by balupton** [reply]
`nu-check` doesn't seem to be useful:

when the script is running:

```
Warning: nu::parser::deprecated

  ! Flag deprecated.
     ,-[/home/balupton/.config/dorothy/commands/screentime:152:41]
 151 |     let input_lower = ($input | str downcase)
 152 |     let alias_result = ($APP_ALIASES | get -i $input_lower)
     :                                            ^|
     :                                             `-- get --ignore-errors was deprecated in 0.106.0 and will be removed in a future release.
 153 |     if $alias_result != null {
     `----
  help: This flag has been renamed to `--optional (-o)` to better reflect its behavior.
```

when I check it:

```
$ nu -c 'nu-check -d /home/balupton/.config/dorothy/commands/screentime'
true
```

**Comment by balupton** [comment]
There is now https://github.com/wvhulle/nu-lint which handles things like types and conventions.

Still nothing that offers deprecation checks.

**Comment by iwinux** [comment]
Without shellcheck-like linter I'm not confident in writing Nu scripts at all.
